Add sub menu of diagram page

HyeongBaeYu opened the pull request:
It looks that huginn needs the convenient web UI for user.
So I attached sub menu on each agent on diagram page.
I hope It make users check/change agent settings easily.
If it will be merged, I will add additional function to sub menu later.

![image](https://cloud.githubusercontent.com/assets/9605660/9733690/5fb60882-5668-11e5-8c89-5199ea725486.png)

HyeongBaeYu committed:
Add sub menu of diagram

cantino commented:
Awesome @HyeongBaeYu, thank you!  I wonder if there is anyway we could add some RSpec coverage on this?  It may not be easy, though.

cantino commented:
@knu wrote much of the original code and may have some code review thoughts.  Nice improvement!

HyeongBaeYu commented:
@cantino  Good to see you again!!  As you say,to add rspec coverage on this change at this update is not easy. I need to study rspec more  :)

But I have plan to add more menu to this sub menu later. I will try to add some rspec coverage with next update.

knu commented:
Wow, this is a lot of work!  I was about to start tweaking the diagram UI, but this may supersede part of what I was thinking of doing.  OK, here's a couple of comments.

- Can't our common agent action menu (`app/views/agents/_action_menu.html.erb`) be used here?  If it is too much to pre-generate a menu for each agent, dynamically generating one via Ajax call on click may be an idea.
- Does this remove the current colored badges?  I'd like to keep them since they let you see at a glance which agents are working or not working.
- Style issue: put a space right before every `<<` operator and after every comment marker (`#`)

Looking forward to seeing this complete!

HyeongBaeYu committed:
Fix style issue and Change background color by agent status

- Fix style issue
- Change background color by agent status
   agent.unavailable case => 'label-warning' color
   agent.working case => 'label-success' color
   else case => 'label-danger'

HyeongBaeYu commented:
Hello @knu 

Thanks for your advice and review.

1. Can't our common agent action menu (app/views/agents/_action_menu.html.erb) be used here? If it is too much to pre-generate a menu for each agent, dynamically generating one via Ajax call on click may be an idea.
=> Yes I don't also want pre-generate with a lot of work. And I tried to reuse  _action_menu.html.erb and add dynamically generating menu on agent click. But It was failed of my lack of ability :). I will have challenge for compact coding later.

2. Does this remove the current colored badges? I'd like to keep them since they let you see at a glance which agents are working or not working.
=> I change background color by agent status

3. Style issue: put a space right before every << operator and after every comment marker (#)
=> I fixed and updated it.

cantino commented:
What are the next steps here?
